packages/assets-controllers/src/NftController.test.ts

```diff
@@ -2874,34 +3108,84 @@ describe('NftController', () => {
-          const { nftController } = setupController();
-          sinon.stub(nftController, 'isNftOwner' as any).returns(false);
+    it('should set NFT as favorite when passed networkClientId and userAddress in options', async () => {
+      const { nftController, preferences, changeNetwork } = setupController();
 
-          const { selectedAddress, chainId } = nftController.config;
-          await nftController.addNft('0x02', '1', {
-            nftMetadata: {
-              name: 'name',
-              image: 'image',
-              description: 'description',
-              standard: 'standard',
-              favorite: false,
-            },
-          });
-
-          expect(
-            nftController.state.allNfts[selectedAddress][chainId][0]
-              .isCurrentlyOwned,
-          ).toBe(true);
-
-          await nftController.checkAndUpdateAllNftsOwnershipStatus();
-          expect(
-            nftController.state.allNfts[selectedAddress][chainId][0]
-              .isCurrentlyOwned,
-          ).toBe(false);
+      const userAddress1 = '0x123';
+      const userAddress2 = '0x321';
+
+      changeNetwork(SEPOLIA);
+      preferences.update({ selectedAddress: userAddress1 });
+
+      await nftController.addNft(
+        ERC721_DEPRESSIONIST_ADDRESS,
+        ERC721_DEPRESSIONIST_ID,
+        { nftMetadata: { name: '', description: '', image: '', standard: '' } },
+      );
+
+      expect(
+        nftController.state.allNfts[userAddress1][SEPOLIA.chainId][0],
+      ).toStrictEqual(
+        expect.objectContaining({
+          address: ERC721_DEPRESSIONIST_ADDRESS,
+          tokenId: ERC721_DEPRESSIONIST_ID,
+          favorite: false,
+        }),
+      );
+
+      changeNetwork(GOERLI);
+      preferences.update({ selectedAddress: userAddress2 });
+
+      // now favorite the nft after changing to a different account from the one where it was added
+      nftController.updateNftFavoriteStatus(
+        ERC721_DEPRESSIONIST_ADDRESS,
+        ERC721_DEPRESSIONIST_ID,
+        true,
+        {
+          networkClientId: SEPOLIA.type,
+          userAddress: userAddress1,
+        },
+      );
+
+      expect(
+        nftController.state.allNfts[userAddress1][SEPOLIA.chainId][0],
+      ).toStrictEqual(
+        expect.objectContaining({
+          address: ERC721_DEPRESSIONIST_ADDRESS,
+          tokenId: ERC721_DEPRESSIONIST_ID,
+          favorite: true,
+        }),
+      );
+    });
+  });
+
+  describe('checkAndUpdateNftsOwnershipStatus', () => {
+    describe('checkAndUpdateAllNftsOwnershipStatus', () => {
+      it('should check whether NFTs for the current selectedAddress/chainId combination are still owned by the selectedAddress and update the isCurrentlyOwned value to false when NFT is not still owned', async () => {
+        const { nftController } = setupController();
+        sinon.stub(nftController, 'isNftOwner' as any).returns(false);
```

Comment:
Yeah so this test just got moved so I didn't actually add this. But you're right this isn't actually necessary. A type error is thrown because `isNftOwner` returns a promise so it should be
```
sinon.stub(nftController, 'isNftOwner' as any).resolves(false)
```

not sure if it makes sense to add this change to this already pretty bloated PR but agreed we should change it. 🤔 not sure why it works now.

Reply:
You can leave it out! As long as we aren't making anything worse haha. Have be careful changing more than necessary with these refactors. There's always something that needs to be fixed after all